src/world.c

```diff
@@ -1191,6 +1191,94 @@ void flecs_default_move_w_dtor(void *dst_ptr, void *src_ptr,
     cl->dtor(src_ptr, count, ti);
 }
 
+/* Define noreturn attribute only for GCC or Clang.
+ * Certain builds in Windows require this for functions that abort 
+ * (-Wmissing-noreturn)
+*/
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
I don't think this is necessary since all functions that use it have a `void` return type. Not major enough to prevent the PR from landing, but probably something I'll end up removing unless it breaks CI.

As an aside, defines like this belong in `include/private/api_defines.h`.

Reply:
I put it due to CI Windows msys builds complaining.

I will move this to the right place.